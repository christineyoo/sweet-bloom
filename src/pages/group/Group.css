#group {
  padding: 2rem;
}

.shop-flex {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.responsive {
  width: 100%;
  height: auto;
}

@media all and (min-width: 800px) {
  #group {
    padding: 5% 20%;
  }
}
