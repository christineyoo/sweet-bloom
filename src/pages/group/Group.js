import React, { Component } from 'react';
import ApiContext from '../../ApiContext';
import FindGroupName from '../../organisms/findGroupName/FindGroupName';
import './Group.css';
import ItemLink from '../../organisms/itemLink/ItemLink';

class Group extends Component {
  static contextType = ApiContext;

  filterItems = () => {
    const copyItems = this.context.items || [];
    return (
      <div className='shop-flex'>
        {copyItems
          .filter((item) => item.group_id === +this.props.match.params.groupId)
          .map((item, i) => {
            return (
              <ItemLink
                key={i}
                item_id={item.id}
                item_name={item.item_name}
                item_price={item.item_price}
                item_url={item.item_url}
                item_description={item.item_description}
              />
            );
          })}
      </div>
    );
  };

  render() {
    return (
      <div id='group'>
        <h1>
          Viewing all{' '}
          <FindGroupName group_id={this.props.match.params.groupId} />
        </h1>
        {this.filterItems()}
      </div>
    );
  }
}

export default Group;
