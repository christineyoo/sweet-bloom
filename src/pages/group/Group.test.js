import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Group from './Group';

it('renders Group UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <Group />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
