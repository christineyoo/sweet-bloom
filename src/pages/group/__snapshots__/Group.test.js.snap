// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Group UI as expected 1`] = `
<div
  id="group"
>
  <h1>
    Viewing all
     
    Desserts
  </h1>
  <div
    className="shop-flex"
  />
</div>
`;
